src/java.base/share/classes/java/security/Security.java

```diff
@@ -141,8 +141,8 @@ private static void initialize() {
         }
 
         /*[IF CRIU_SUPPORT]*/
-        // Check if CRIU checkpoint mode is enabled, if it is then reconfigure the security providers.
-        if (InternalCRIUSupport.isCheckpointAllowed()) {
+        // If -XX:-CRIUSecProvider is specified, we don't need to configure the CRIUSec provider.
+        if (InternalCRIUSupport.isCheckpointAllowed() && InternalCRIUSupport.enableCRIUSecProvider()) {
```

Comment:
Please keep the comment the same as in the head stream.

Reply:
updated